src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -166,4 +175,13 @@ public boolean checkVelocity() {
                 leftEncoder.getVelocity() - ShooterConstants.BOTTOM_MOTOR_SETPOINT_SPEAKER.in(RPM))
             <= ShooterConstants.SHOOTER_RANGE;
   }
+
+  public void ShooterRest() {
+    rightController.setOutputRange(0.0, 0.0);
```

Comment:
You're using setOutputRange in the setVelocity and here to turn off the output.
Generally, you do not want to change configuration on the controllers during a match. This also has the effect that the PID loop is still running the entire time it is unable to provide any power.

The idea here is that the motor is enabled when shooting and disabled when it is not. The easiest way to turn off power to a motor is 
.setReference(0.0, ControlType.kVoltage)

The shooter has exactly one velocity. This can be set from a constant on construction. So you don't need a setVelocity command or method.